bfvmm/include/vmcs/vmcs_intel_x64.h

```diff
@@ -455,142 +455,109 @@ class vmcs_intel_x64
 };
 
 inline auto
-get_vmcs_field(uint64_t addr, const std::string &name, bool exists)
+get_vmcs_field(uint64_t addr, const char *name, bool exists)
 {
-    auto what = std::string("get_vmcs_field failed: ") + name + " field doesn't exist";
-
     if (!exists)
-        throw std::logic_error(what);
+        throw std::logic_error("get failed: field doesn't exist");
```

Comment:
update this to include the name. It's safe to use the "+" here as the throw should never happen. The old code was creating the "what" all the time, which is why it was an issue. The normal case needs to be as fast as possible. The error case can be slow.